contracts/Pricing.sol

```diff
@@ -267,10 +267,10 @@ contract Pricing is IPricing {
         uint256 underlyingInstances = 0;
         for (uint8 i = 0; i < 8; i++) {
             int256 timeWeight = 8 - i;
-            uint256 j = currentHour - i; // keep moving towards 0
+            int256 j = int256(currentHour) - int256(i); // keep moving towards 0
             // loop back around list if required
             if (j < 0) {
-                j = 24 + (currentHour - i);
+                j = 23;
```

Comment:
If currentHour != i - 1, j might not necessarily equal 23.

On phone at the moment, so can't check properly for another couple of hours, but just wanted to make sure this was known.

Reply:
@BenjaminPatch Jack and I whiteboarded this yesterday arvo and the conclusion we came to was that currentHour is in [0,23]. And in this snippet the `if j < 0` case should always be called when j = -1, since j will start at current hour and move towards 0.

It seemed to us that having 24 + (currentHour - i) then meant that since this always gets triggered when j (which is equal to currentHour - 1 anyway) is equal to -1, then this should always be a constant 23. Thoughts?